### PR TITLE
Add SamuraiHudWindow

### PR DESCRIPTION
**Pull request by abazilla**
Had a go at adding Samurai to the list of HUDs! Please give me some feedback - it's both my first time in C# and contributing to open source :) 

gif: https://imgur.com/XN6bqJr

### CODE REVIEW

File: Interface/SamuraiHudWindow.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            DrawSenResourceBar();
+            DrawMeditationResourceBar();
+            DrawTargetBar();
+        }
+
+        protected override void DrawPrimaryResourceBar()
+        {
+            var gauge = PluginInterface.ClientState.JobGauges.Get<SAMGauge>();
+
+            // const int xPadding = 5;
```
Comment (jdsmith2816) [inline Interface/SamuraiHudWindow.cs:30]: Let's try to keep commented out code out of the repo;  everything is in source control so we can always bring it back if we need it.

File: Interface/SamuraiHudWindow.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            
+            var barWidth = (BarWidth - xPadding * (numChunks - 1)) / numChunks;
+            var barSize = new Vector2(barWidth, BarHeight);
+            var xPos = CenterX - XOffset;
+            var yPos = CenterY + YOffset ;
+            var cursorPos = new Vector2(xPos - xPadding - barWidth, yPos);
+            
+            var drawList = ImGui.GetWindowDrawList();
+            
+            for (var i = 1; i < 4; i++) {
```
Comment (jdsmith2816) [inline Interface/SamuraiHudWindow.cs:79]: Why are you looping over this? I don't think we're gaining anything in clarity or brevity here. 

Personally I think we remove the loop and the `if i == #` checks and just have the duplicated cursorpos and border rect lines.

File: Interface/SamuraiHudWindow.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                cursorPos = new Vector2(cursorPos.X + xPadding + barWidth, cursorPos.Y);
+                
+                if (i == 1)
+                {
+                    if (gauge.HasSetsu())
+                    {
+                        drawList.AddRectFilled(
+                        cursorPos, cursorPos + barSize,
+                        ImGui.ColorConvertFloat4ToU32(new Vector4(89 / 255f, 234 / 255f, 247 / 255f, 255f))
```
Comment (jdsmith2816) [inline Interface/SamuraiHudWindow.cs:88]: IMO we should just directly put the color here.   If you google `rgb to hex` you'll get a little widget with an RGB box on the bottom left that you can type `89 234 247` into and it'll show you the HEX value of `#59eaf7` 

Now that you know the hexcode you can replace this line of code with 0xAABBGGRR  AA is FF for 100% opacity so your color becomes.. 0xFFF7EA59

Repeat this process for the other three colors.

**Review by jdsmith2816 — CHANGES_REQUESTED** [review]
This mostly looks fine to me other than the few comments I left.  Great first PR!

One final note... Let's try to keep the commitlog relatively clean.  When you update the PR for needed changes go ahead and squash down to one commit with a message like

`feat: add samurai support with custom job gauges`

**Comment by abazilla** [comment]
All done! :) thanks for the review.

**Comment by jdsmith2816** [comment]
Thanks abazilla!

**Pull request merged**